pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -774,15 +778,32 @@ Set<Integer> getPartitionIds(StreamConfig streamConfig)
     }
   }
 
+  @VisibleForTesting
+  Set<Integer> getPartitionIds(List<StreamConfig> streamConfigs)
+      throws Exception {
+    Set<Integer> partitionIds = new HashSet<>();
+    for (int i = 0; i < streamConfigs.size(); i++) {
+      final int index = i;
+      try {
+        partitionIds.addAll(getPartitionIds(streamConfigs.get(index)).stream().map(
```

Comment:
can you please elaborate on why we use unordered set to store partitionids?

Reply:
This function is overloading an existing same name function. The other one is also returning a Set<>. Usage of the output is only checking if partitionId exists instead of checking its order.